### PR TITLE
support snarkjs

### PR DESCRIPTION
**Pull request by zhfnjust**

### CODE REVIEW

File: zh/course-02/lesson-03/chapter-01/README.md [hunk 1]
```diff
@@ -1,41 +1,147 @@
...
+然后使用以下命令从安装我们的 [扩展版本](https://github.com/sCrypt-Inc/snarkjs)：
+
+```
+npm install https://github.com/sCrypt-Inc/snarkjs.git
+```
+
+
+## 工作流程
+
+整个工作流程与原始 snarkjs 相同，只是验证步骤是在比特币上完成的。一般来说，它包含以下步骤：
```
Comment (xhliu) [inline zh/course-02/lesson-03/chapter-01/README.md:27]: Add "Visual Summary" https://docs.circom.io/

File: zh/course-02/lesson-03/chapter-02/README.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+如上所述，我们可以使用 `(pos_x, pos_y,orientation)` 的元组来表示一艘船在网格中的位置，但是我们如何表示整个舰队的位置状态呢？
+
+请注意，位置的有效值在 `0` 到 `9` 之间，方向值只是 `0` 或 `1`。所以我们可以将它们中的每一个编码为 `4` 个比特位，并将它们全部连接起来形成一个可以表示整个元组的数字。例如：
+
+```
+var val =  ships[i][0] + (ships[i][1] * 16) + (ships[i][2] * 16 * 16);
+```
+
+用同样的想法，我们可以在一个“字段”中表示整个舰队的位置状态：
```
Comment (xhliu) [inline zh/course-02/lesson-03/chapter-02/README.md:20]: “字段” -> 变量

Change in Zokrates:
“字段”
->
field

**Review by xhliu — APPROVED** [review]
👍 well done.


Some minor changes. Get the English version next.

File: zh/course-02/lesson-03/chapter-01/README.md [hunk 3]
```diff
@@ -1,41 +1,147 @@
...
+circom factor.circom --r1cs --wasm
 ```
-zokrates export-verifier-scrypt
+
+### 3. 开始新的权力tau仪式
+
+这个 `new` 的命令用于启动 权力 tau 的仪式。
+
+```bash
+snarkjs powersoftau new bn128 12 pot12_0000.ptau
```
[the remark this answers is not in the document]
Comment (zhfnjust) [reply zh/course-02/lesson-03/chapter-01/README.md:70]: it 's a output file

**Pull request merged**